### ISSUE TITLE
Override channel option not functioning

### ISSUE THREAD

**Issue by mgreensmith** [ashwanthkumar/gocd-slack-build-notifier#22]
Adding a `channel` override to a pipeline rule does not seem to work. Notifications for all rules are being sent to the primary `gocd.slack.channel` if configured, and if it is not configured, are being sent to the default channel configured in the slack webhook integration.

Given a slack webhook integration with a default channel of `#general`, and this partial config:

```
gocd.slack {
  channel = "#gocd"
  pipelines = [{
    name = "foo"
    channel = "#foo"
  }]
}
```

I would expect notifications for pipeline `foo` to be sent to `#foo` Slack channel. However, all notifications for pipeline `foo` are being sent to the `#gocd` Slack channel. If I remove the `channel = "#gocd"` configuration option, all notifications for pipeline `foo` are sent to the default `#general` channel.

**Comment by ashwanthkumar** [comment]
@mgreensmith Can you please also attach the logs for the plugin? I'm specifically looking for log lines that says "Updating target slack channel to <slack channel>". This will be printed only on DEBUG mode.

**Comment by ashwanthkumar** [comment]
Also if possible try running this version (https://github.com/ashwanthkumar/gocd-slack-build-notifier/releases/tag/v1.1.1) of the plugin. This contains additional logs in DEBUG mode that would emit the pipeline rule that's being fired.

**Comment by mgreensmith** [comment]
@ashwanthkumar I just tried to reproduce the bug using `v1.1.1` of the plugin, and to my surpise, I couldn't repro -- everything worked perfectly, and messages were posted to the correct channels. 

After looking through commit history, it looks like this bug has already been fixed via https://github.com/ashwanthkumar/gocd-slack-build-notifier/commit/095ab2cdcbe4e1f02bb9bb92fa7fe78a3d088bfe, but I was running `v1.0` locally and did not have that patch applied. So, thanks for releasing this new version of the plugin!
